src/Components/ManageTeam/EmployeeForm.tsx

```diff
@@ -0,0 +1,108 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Button, FormControl, FormHelperText, InputLabel, MenuItem, Select, TextField } from '@mui/material';
+import { useEffect, useState } from 'react';
+import { Controller, SubmitHandler, useForm } from 'react-hook-form';
+import * as Yup from 'yup';
+
+type NewEmployeeData = {
+  login: string;
+  password: string;
+  role: string;
+};
+
+const schema = Yup.object().shape({
+  login: Yup.string().required('Required'),
+  password: Yup.string().required('Required'),
+  role: Yup.string().required('Required'),
+});
+
+type EmployeeFormProps = {
+  recordForEdit: NewEmployeeData | null;
+  addOrEdit: () => void;
+};
+
+export function EmployeeForm({ recordForEdit, addOrEdit }: EmployeeFormProps) {
+  const [data, setData] = useState<NewEmployeeData>({
```

Comment:
Could be a more specific name to know just by looking at any part of the code what kind of data it is.